File: Node/src/dto/objects/Statistic.ts
import { ValidateNested } from "class-validator"
import { Price } from "../primitives/Price"
import { Int } from "../tsPrimitives/Int"
import { UUID } from "crypto"
import { IsoDate } from "../tsPrimitives/IsoDate"

export namespace Statistic {

    export namespace Parameters {

        export class AppointmentsQuery {
            startDate: IsoDate
            endDate: IsoDate

            @ValidateNested()
            employees?: UUID[]
            
            @ValidateNested()
            salons?: UUID[]

            constructor(
                startDate: IsoDate,
                endDate: IsoDate,
                employees?: UUID[],
                salons?: UUID[]
            ) {
                this.startDate = startDate
                this.endDate = endDate
                this.employees = employees
                this.salons = salons
            }
        }
    }

    export namespace Responses {

        export class Appointments {
            @ValidateNested()
            price: Price

            @ValidateNested()
            count: Int

            constructor(
                price: Price, 
                count: Int
            ) {
                this.price = price
                this.count = count
            }
        }
    }
}

File: IOS/DTOs/Objects/Employee/Statistic.swift
import Foundation
import MemberwiseInit

public enum Statistic {
    public enum Parameters {}
    public enum Responses {}
}

public extension Statistic.Parameters {
    
    @MemberwiseInit(.public, _optionalsDefaultNil: true)
    struct AppointmentsQuery: Codable {
        public let startDate: Date
        public let endDate: Date
        public let employees: [UUID]?
        public let salons: [UUID]?
    }
}

public extension Statistic.Responses {
    
    @MemberwiseInit(.public, _optionalsDefaultNil: true)
    struct Appointments: Codable {
        public var price: Price
        public var count: Int
    }
}
